cli/how-to-train-autorest.py

```diff
@@ -0,0 +1,160 @@
+docker_spec = DockerImage(
+    docker_image_uri="mcr.microsoft.com/azureml/openmpi3.1.2-ubuntu18.04"
+)
+properties = EnvironmentSpecificationVersion(conda_file=conda_file, docker=docker_spec)
+request = EnvironmentSpecificationVersionResource(properties=properties)
+env_version = client.environment_specification_versions.create_or_update(
+    "lightgbm-environment",
+    1,
+    subscription_id=subscription_id,
+    workspace_name=workspace_name,
+    resource_group_name=resource_group_name,
+    body=request,
+)
+# </create environment>
+
+print(f"Created environment: {env_version}")
+
+# <create data>
+properties = DataVersion(
+    dataset_type="simple",
+    path="https://azuremlexamples.blob.core.windows.net/datasets/iris.csv",
+)
+request = DataVersionResource(properties=properties)
+data_version = client.data_versions.create_or_update(
+    "iris-data",
+    1,
+    body=request,
+    subscription_id=subscription_id,
+    workspace_name=workspace_name,
+    resource_group_name=resource_group_name,
+)
+# </create data>
+
+print(f"Created data version: {data_version}")
+
+# Create Code
+# TODO: decide how to upload to container
+
+datastores = client.datastores.list(
+    subscription_id=subscription_id,
+    workspace_name=workspace_name,
+    resource_group_name=resource_group_name,
+)
+datastore_id = list(filter(lambda d: d.name == "workspaceblobstore", datastores))[0].id
+print(f"Using datastore: {datastore_id}")
+
+# <create code>
+properties = CodeVersion(datastore_id=datastore_id, path="src")
+request = CodeVersionResource(properties=properties)
+code_version = client.code_versions.create_or_update(
+    "train-lightgbm",
+    1,
+    body=request,
+    subscription_id=subscription_id,
+    workspace_name=workspace_name,
+    resource_group_name=resource_group_name,
+)
+# </create code>
+
+print(f"Created code version: {code_version}")
+
+# TODO: figure out get compute
+# <create compute binding>
+compute_name = "cpu-cluster"
+compute_id = f"/subscriptions/{subscription_id}/resourceGroups/{resource_group_name}/providers/Microsoft.MachineLearningServices/workspaces/{workspace_name}/computes/{cluster}"
+compute_binding = ComputeConfiguration(
+    target=compute_id,
+    instance_count=1,
+)
+# </create compute binding>
+
+# <create data binding>
+data_config = {
+    "iris": InputDataBinding(
+        data_id=data_version.id,
+    )
+}
+# </create data binding>
+
+# <create job>
+properties = CommandJob(
+    code_id=code_version.id,
+    experiment_name="lightgbm-iris",
+    environment_id=env_version.id,
+    command="python main.py --iris-csv $AZURE_ML_INPUT_iris",
+    input_data_bindings=data_config,
+    compute=compute_binding,
+)
+
+request = JobBaseResource(properties=properties)
+job_id = uuid.uuid4()
```

Comment:
can we make it inline? reducing number of lines?

Reply:
No, I use this again for `client.jobs.get`. We could remove it but then have to extract this from the full resource id which I think is a mistake